src/zenml/integrations/xgboost/materializers/xgboost_booster_materializer.py

```diff
@@ -0,0 +1,66 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import os
+import tempfile
+from typing import Any, Type
+
+import xgboost as xgb
+
+from zenml.artifacts import ModelArtifact
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+DEFAULT_FILENAME = "model.json"
+
+
+class XgboostBoosterMaterializer(BaseMaterializer):
+    """Materializer to read data to and from xgboost.Booster."""
+
+    ASSOCIATED_TYPES = (xgb.Booster,)
+    ASSOCIATED_ARTIFACT_TYPES = (ModelArtifact,)
+
+    def handle_input(self, data_type: Type[Any]) -> xgb.Booster:
```

Comment:
Got one small complaint for all four materializer methods: This will create (potentially large) temporary files that won't be deleted automatically. Can we maybe change that to either a context manager that deletes them or having some cleanup before returning?